test/regression/riscv/instructions/compressed.cc

```diff
@@ -0,0 +1,692 @@
+  EXPECT_EQ(getFPRegister<double>(10), -0.00032);
+  EXPECT_EQ(getFPRegister<double>(11), 123456);
+}
+
+TEST_P(InstCompressed, addi4spn) {
+  // Add immediate to stack pointer
+  RUN_RISCV_COMP(R"(
+    c.addi4spn x8, x2, 4
+    c.addi4spn x9, x2, 12
+  )");
+  EXPECT_EQ(getGeneralRegister<uint64_t>(8), process_->getStackPointer() + 4);
+  EXPECT_EQ(getGeneralRegister<uint64_t>(9), process_->getStackPointer() + 12);
+}
+
+TEST_P(InstCompressed, sw) {
+  // Compressed store word
+  initialHeapData_.resize(16);
+  uint32_t* heap = reinterpret_cast<uint32_t*>(initialHeapData_.data());
+  heap[0] = 0x12345678;
+  heap[1] = 0xDEADBEEF;
+  heap[2] = 0x87654321;
+
+  RUN_RISCV_COMP(R"(
+      # Get heap address
+      li a7, 214
+      ecall
+
+      li x8, 0xAA
+      c.sw x8, 0(a0)
+
+      addi x8, x8, 0xAA  # 0xAA + 0xAA = 154
+      slli x8, x8, 16
+      addi x8, x8, 0xAA  # 0x15400AA
+      c.sw x8, 4(a0)
+  )");
+
+  EXPECT_EQ(getGeneralRegister<uint64_t>(10), 32);
+  EXPECT_EQ(getMemoryValue<uint64_t>(32), 0x015400AA000000AA);
+  EXPECT_EQ(getMemoryValue<uint64_t>(36), 0x87654321015400AA);
+}
+
+TEST_P(InstCompressed, sd) {
+  // Compressed store double word
+  initialHeapData_.resize(16);
+  uint32_t* heap = reinterpret_cast<uint32_t*>(initialHeapData_.data());
+  heap[0] = 0x12345678;
+  heap[1] = 0xDEADBEEF;
+  heap[2] = 0x87654321;
+
+  RUN_RISCV_COMP(R"(
+      # Get heap address
+      li a7, 214
+      ecall
+
+      li x8, 0xAA
+      c.sd x8, 0(a0)
+
+      addi x8, x8, 0xAA  # 0xAA + 0xAA = 154
+      slli x8, x8, 16
+      addi x8, x8, 0xAA  # 0x15400AA
+      c.sd x8, 8(a0)
+  )");
+
+  EXPECT_EQ(getGeneralRegister<uint64_t>(10), 32);
+  EXPECT_EQ(getMemoryValue<uint64_t>(32), 0x00000000000000AA);
+  EXPECT_EQ(getMemoryValue<uint64_t>(40), 0x00000000015400AA);
+}
+
+TEST_P(InstCompressed, fsd) {
+  // Compressed store double precision float
+  initialHeapData_.resize(32);
+  double* heap = reinterpret_cast<double*>(initialHeapData_.data());
+  heap[0] = 1.0;
+  heap[1] = 123.456;
+  heap[2] = -0.00032;
+  heap[3] = 123456;
+
+  RUN_RISCV_COMP(R"(
+     # Get heap address
+     li a7, 214
+     ecall
+
+     fld fa0, 0(a0)
+     fld fa1, 8(a0)
+     fld fa2, 16(a0)
+     fld fa3, 24(a0)
+
+     c.fsd fa3, 0(a0)
+     c.fsd fa2, 8(a0)
+     c.fsd fa1, 16(a0)
+     c.fsd fa0, 24(a0)
+   )");
+
+  EXPECT_EQ(getFPRegister<double>(10), 1.0);
+  EXPECT_EQ(getFPRegister<double>(11), 123.456);
+  EXPECT_EQ(getFPRegister<double>(12), -0.00032);
+  EXPECT_EQ(getFPRegister<double>(13), 123456);
+
+  EXPECT_EQ(getGeneralRegister<uint64_t>(10), 32);
+
+  EXPECT_EQ(getMemoryValue<double>(32), 123456);
+  EXPECT_EQ(getMemoryValue<double>(40), -0.00032);
+  EXPECT_EQ(getMemoryValue<double>(48), 123.456);
+  EXPECT_EQ(getMemoryValue<double>(56), 1.0);
+}
+
+TEST_P(InstCompressed, j) {
+  // Compressed jump
+  // Labels needed as LLVM eagerly uses compressed instructions e.g. addi ->
+  // c.addi causing manual jump offsets to be confusing
```

Comment:
I think "to be confusing" should be reworded to something more explanatory/definite. Something like "to be more complex to calculate" or "to be less feasible to calculate". Something along those lines